tests/unit_tests/http.cpp

```diff
@@ -250,13 +256,13 @@ std::string get_nc(std::uint32_t count)
 
 TEST(HTTP_Server_Auth, NotRequired)
 {
-  http::http_server_auth auth{};
+  http::http_server_auth auth{rng};
```

Comment:
Forgot to mention to drop this change. The test will then verify that nothing bad happens when `std::function` is not set. It throws in that situation FWIW.